### DISCUSSION TITLE
New to Yoshi Engine and idk what im doing

### DISCUSSION THREAD

**Discussion by Mysterious-Madness** [YoshiCrafter29/YoshiCrafterEngine#131] [Q&A]
Q: How to create function events
- The stuff is REALLY confusing

**Comment by BernardoGP4504** [comment]
`function your-event-function(Some string of event1:String, Some string of event2:String, Some string of event3:String) {`
"your-event-function" being the function name you give it (can be named literally anything)
and "Some string of event(number that i put there) being the fields the user has to fill when using the event (you also name it whatever you want), and it can go even further with those.
after you put inside the function (new line with 1 [Tab] pressed after that function thing) make sure to create a new line and put "}" without quotes in there (and erase the space of that 1 tab for it to be aligned at the start of the line)
for example, my broken character switch event:
```
function characterSwitch(Character:String, xPos:String, yPos:String){
    PlayState.remove(Playstate.boyfriend);
    Playstate.boyfriend.destroy();
    Playstate.boyfriends = [new Boyfriend(Std.parsefloat(xPos), Std.parsefloat(yPos), character)];
    Playstate.add(Playstate.boyfriend);
    Playstate.iconP1.changeCharacter(character, mod);
}
```
